java.base/src/main/java/java/lang/reflect/Array$_native.java

```diff
@@ -38,4 +38,36 @@ private static Object newArray(Class<?> componentType, int length) throws Negati
             return CompilerIntrinsics.emitNewReferenceArray(componentType, dimensions, length);
         }
     }
+
+    private static Object multiNewArray(Class<?> componentType, int[] dimensions) throws IllegalArgumentException, NegativeArraySizeException {
+        int dimCount = dimensions.length;
+        if (componentType.isPrimitive()) {
+            dimCount -= 1;
+        } else {
+            while (componentType.isArray()) {
+                dimCount += 1;
+                componentType = componentType.getComponentType();
+            }
```

Comment:
This is not something we have to fix now, but it should be possible to get the dimensions and leaf type directly from any reference array class thus avoiding a loop.